dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -346,10 +354,22 @@ func (rrh *resolverRequestHandler) RequestTrieNodes(destShardID uint32, hashes [
 	if len(unrequestedHashes) == 0 {
 		return
 	}
+
+	rrh.mutexTrieHashes.Lock()
+	defer rrh.mutexTrieHashes.Unlock()
+
+	rrh.trieHashAccumulator = append(rrh.trieHashAccumulator, unrequestedHashes...)
+	elapsedTime := time.Since(rrh.lastTrieRequestTime)
+	if len(rrh.trieHashAccumulator) < minHashesToRequest && elapsedTime < timeToAccumulateTrieHashes {
+		return
+	}
+
+	rrh.lastTrieRequestTime = time.Now()
```

Comment:
Move this to line 391 where you are sure that the rrh.requestHashesWithDataSplit(rrh.trieHashAccumulator, trieResolver) would be executed

Reply:
I would make requestHashesWithDataSplit to return error at least if the SplitDataInChunks call fails. And set the rrh.lastTrieRequestTime = time.Now() only if the request has been done with success.

Reply:
done.